CelesteNet.Server.ChatModule/CMDs/CmdLocate.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using Celeste.Mod.CelesteNet.DataTypes;
+using MonoMod.Utils;
+
+namespace Celeste.Mod.CelesteNet.Server.Chat.Cmd;
+
+public class CmdLocate : ChatCmd {
+
+    public override CompletionType Completion => CompletionType.Player;
+
+    public override string Info => "Find where a player is.";
+
+    private CelesteNetPlayerSession? Other;
+    private DataPlayerInfo? OtherPlayer;
+
+    public override void Init(ChatModule chat) {
+        Chat = chat;
+
+        ArgParser parser = new(chat, this);
+        parser.AddParameter(new ParamPlayerSession(chat, ValidatePlayerSession));
+        ArgParsers.Add(parser);
+    }
+
+    private void ValidatePlayerSession(string raw, CmdEnv env, ICmdArg arg) {
+        if (arg is not CmdArgPlayerSession sessionArg)
+            throw new CommandRunException("Invalid username or ID.");
+
+        CelesteNetPlayerSession? other = sessionArg.Session;
+        DataPlayerInfo otherPlayer = other?.PlayerInfo ?? throw new CommandRunException("Invalid username or ID.");
+
+        Other = other;
+        OtherPlayer = otherPlayer;
+    }
+
+
+    public override void Run(CmdEnv env, List<ICmdArg>? args) {
+        if (Other == null || OtherPlayer == null)
+            throw new InvalidOperationException("This should never happen if ValidatePlayerSession returns without error.");
+
+        CelesteNetPlayerSession? self = env.Session ?? throw new CommandRunException("Cannot locate as the server.");
+
+        var chat = new DataChat {
+            Player = OtherPlayer,
+            Tag = "locate",
+            Text = "If you see this, either your client does not properly handle /locate or this is a bug. Please report!",
```

Comment:
Considering this is what players on outdated clients will see, and people don't tend to update their mods a lot of the time, I'm not sure if I'm that big of combining the two cases (outdated client / unhandled locate packet) into one - ideally, support for this command would be locked behind a handshake feature, so it might be worth looking into adding that to the PR.

Reply:
I'll look into it!

Reply:
See my latest commit.